openapiart/bundler.py

```diff
@@ -863,224 +863,323 @@ def _generate_value_schema(
+                    "start": {
+                        "type": type_name,
+                        "x-field-uid": counter_auto_field.uid,
                     },
-                    "offset": {
-                        "description": "Offset in bits relative to start of corresponding header field",
-                        "type": "integer",
-                        "format": pattern_fmt,
-                        "default": 0,
-                        "maximum": length - 1,
-                        "x-field-uid": metric_tags_auto_field.uid,
+                    "step": {
+                        "type": type_name,
+                        "x-field-uid": counter_auto_field.uid,
                     },
-                    "length": {
-                        "description": "Number of bits to track for metrics starting from configured offset of corresponding header field",
+                    "count": {
                         "type": "integer",
-                        "format": pattern_fmt,
-                        "default": length,
-                        "minimum": 1,
-                        "maximum": length,
-                        "x-field-uid": metric_tags_auto_field.uid,
+                        "x-field-uid": counter_auto_field.uid,
                     },
                 },
             }
-            schema["properties"]["metric_tags"] = {
-                "description": "One or more metric tags can be used to enable tracking portion of or all bits in a corresponding header field for metrics per each applicable value. These would appear as tagged metrics in corresponding flow metrics.",
-                "type": "array",
-                "items": {
-                    "$ref": "#/components/schemas/{}".format(
-                        metric_tags_schema_name
-                    )
-                },
-                "x-field-uid": auto_field.uid,
-            }
+            self._apply_common_x_field_pattern_properties(
+                counter_schema["properties"]["start"],
+                xpattern,
+                fmt,
+                property_name="start",
+            )
+            self._apply_common_x_field_pattern_properties(
+                counter_schema["properties"]["step"],
+                xpattern,
+                fmt,
+                property_name="step",
+            )
+            self._apply_common_x_field_pattern_properties(
+                counter_schema["properties"]["count"],
+                xpattern,
+                pattern_fmt,
+                property_name="count",
+            )
+
+            if xconstants is not None:
+                counter_schema["x-constants"] = copy.deepcopy(xconstants)
+
             self._content["components"]["schemas"][
-                metric_tags_schema_name
-            ] = metric_tags_schema
+                counter_pattern_name
+            ] = counter_schema
 
-        # TODO; temporary hack to add auto hierarchy field at the end, remove later
```

Comment:
make sure we cover following scenario before removing TODO
1. Test should not regress, expecting you have some test.
2. I couldn't figure out proper fix of this todo. Could you please pointing out that change.

Reply:
Earlier the fix was to add new auto parameters to end by default forcefully , now the solution is that the features are added according to their position in the features list